src/Castle.Windsor/Compatibility/ApplicationDomains.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+
+namespace Castle.Core.Internal
+{
+    internal static class ApplicationDomains
```

Comment:
I'd prefer them to be duplicated than a shared project, otherwise you could just include the 2 files from the Castle.Windsor directory.